pytorch_lightning/trainer/train_loop_mixin.py

```diff
@@ -423,10 +424,16 @@ def training_forward(self, batch, batch_nb, opt_idx, hiddens):
         # ---------------
         # FORWARD
         # ---------------
-        # enable not needing to add opt_idx to training_step
+        # training_step has an optional optimizer_idx argument
         args = [batch, batch_nb]
         if len(self.optimizers) > 1:
-            args.append(opt_idx)
+            if self.has_arg('training_step', 'optimizer_idx'):
+                args.append(opt_idx)
```

Comment:
I think that it would be safer to move from `args` to `kwargs = {'batch': batch, 'batch_nb': batch_nb}` as args order is not checked properly...

Reply:
ok. this pr or in new one? I suspect that this appears also in other places.

Reply:
it is just my personal feeling, but rather ask @williamFalcon if he agrees too

Reply:
@awaelchli different PR